manager/state/raft/raft.go

```diff
@@ -821,7 +821,14 @@ func (n *Node) ResolveAddress(ctx context.Context, msg *api.ResolveAddressReques
 }
 
 func (n *Node) getLeaderConn() (*grpc.ClientConn, error) {
-	leader := n.Leader()
+	n.stopMu.Lock()
+	if !n.IsMember() {
+		n.stopMu.Unlock()
+		return nil, ErrNoRaftMember
+	}
+
+	leader := n.Node.Status().Lead
+	n.stopMu.Unlock()
```

Comment:
This can just be `leader := n.Leader()` unless you care about returning `ErrNoRaftMember` instead of `"no leader found"`.

I'd rather keep this part of the code as simple as possible. If the returned error matters, maybe we should change `Leader` to return an error. I think there's another place where we reimplement it to distinguish between different error conditions.